backend/src/data/sessionsDataAccess.ts

```diff
@@ -0,0 +1,53 @@
+import { pool } from '../config/db'
+import { v4 as uuidv4 } from 'uuid';
+import { ISession } from '../types/session';
+import { ISessionAccessCode } from '../types/sessionAccessCode';
+
+export const createSession = async (): Promise<ISession | null> => {
+    try {
+   
+  
+      const query = `
+        INSERT INTO sessions (id)
+        VALUES ($1) RETURNING id, active, match_found, created_at;
+      `;
+  
+      const data = [
+        uuidv4()
+      ];
+  
+      const result = await pool.query(query, data);
+  
+      return result.rows[0] || null;
+    } catch (error) {
+      console.log('Error creating session:', error);
+      throw 'Error creating session.';
```

Comment:
if the purpose of this catch is to log the error, I would throw the error again instead of new error

Reply:
This is an area I've always been fuzzy about. My implementation is half baked here and not obvious due to incompleteness, but here's my logic:

The "Throw" is a browser-friendly error message. That message would be OK to be returned back to the client without risk of offering too many details that may be given by the actual error message.

The console.log is just for dev purposes, so I know what's going on in greater detail. I'd either remove it in prod, or add a flag where it only runs if logging=true in .env.

If that makes sense, I'll go ahead and add a flag to all the console.log statements in catch, and ensure they only print when the .env flag "logging" is set to true.

Reply:
Makes sense 👍